src/routes/shopping_cart.py

```
@@ -72,9 +72,11 @@
     try:
         await service.add_movie_to_cart(current_user, movie_id)
         cart = await service.get_cart(current_user)
-        return CartResponse.from_orm(cart)
+        print(f"DEBUG: cart.items = {[item.movie for item in cart.items]}")
+        return CartResponse.from_dict(cart)
     except ValueError as e:
-        raise HTTPException(status_code=400, detail=str(e))
+        status = 404 if "not found" in str(e).lower() else 400
+        raise HTTPException(status_code=status, detail=str(e))
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)